t/admin/upstream.t

```diff
@@ -782,3 +782,94 @@ GET /t
 passed
 --- no_error_log
 [error]
+
+
+
+=== TEST 24: set upstream(type: chash)
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/upstreams/1',
+                 ngx.HTTP_PUT,
+                 [[{
+                    "key": "server_name",
+                    "nodes": {
+                        "127.0.0.1:8080": 1
+                    },
+                    "type": "chash"
+                }]]
+                )
+
+            ngx.status = code
+            ngx.say(body)
+        }
+    }
+--- request
+GET /t
+--- response_body
+passed
+--- no_error_log
+[error]
+
+
+
+=== TEST 25:  wrong upstream key
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/upstreams/1',
+                ngx.HTTP_PUT,
+                [[{
+                    "nodes": {
+                        "127.0.0.1:8080": 1,
+                        "127.0.0.1:8081": 2
+                    },
+                    "type": "chash",
+                    "key": "not_support",
+                    "desc": "new upstream"
+                }]]
+                )
+
+            ngx.status = code
+            ngx.print(body)
+        }
+    }
+--- request
+GET /t
+--- error_code: 400
+--- response_body
+{"error_msg":"invalid configuration: invalid \"pattern\" in docuement at pointer \"#\/key\""}
+--- no_error_log
+[error]
+
+
+
+=== TEST 24: set upstream with args(type: chash)
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/upstreams/1',
+                 ngx.HTTP_PUT,
+                 [[{
+                    "key": "arg_device_id",
```

Comment:
Is this key realy works? you need add e2e(end to end) test cases for that.

Reply:
it  does works,`arg_device_id` means get `device_id` from  queryString

Reply:
you not got my point. 
You set the rule successfully, however, will the request be processed according to the rules you set?

Reply:
for example: https://github.com/iresty/apisix/blob/master/t/node/chash-balance.t

Reply:
I think we can write some debug log to print the `key`: 

https://github.com/iresty/apisix/blob/master/lua/apisix/balancer.lua#L158

and then we can check the content of error log to confirm if the balancer works fine with new `key`.